lib/rubygems/installer.rb

```diff
@@ -766,23 +766,23 @@ def app_script_text(bin_file_name)
 
 require 'rubygems'
 #{gemdeps_load(spec.name)}
-version = "#{Gem::Requirement.default_prerelease}"
-
-str = ARGV.first
-if str
-  str = str.b[/\\A_(.*)_\\z/, 1]
-  if str and Gem::Version.correct?(str)
-    #{explicit_version_requirement(spec.name)}
-    ARGV.shift
+"#{Gem::Requirement.default_prerelease}".tap { |version|
+
+  if ARGV.first =~ /\\A_(.*)_\\z/
+    str = $1
+    if Gem::Version.correct?(str)
```

Comment:
This code is not strictly equivalent to the original code. I don't know with certainty that it would cause any problem, but this is a critical path for almost everything in ruby.